File: python/0001.py
from leetcode_tester import Tester

from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i, n in enumerate(nums):
            if n in d:
                return [d[n], i]
            d[target - n] = i


if __name__ == '__main__':
    solution = Solution()
    test = Tester(solution.twoSum)

    test.addTest(
        [2, 7, 11, 15], 9,
        [0, 1]
    )
    test.addTest(
        [3, 2, 4], 6,
        [1, 2]
    )
    test.addTest(
        [3, 3], 6,
        [0, 1]
    )
    test.doTest()

File: c++/0001.cpp
class Solution {
   public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;

        for (int i = 0; i < nums.size(); i++) {
            int e = target - nums[i];
            if (mp.contains(e)) {
                return {i, mp[e]};
            }
            mp[nums[i]] = i;
        }
        return {0, 0};
    }
};
